scripts/generate-api-protos.sh

```diff
@@ -0,0 +1,195 @@
+  PROTOS=()
+  while IFS= read -r -d '' proto; do
+    PROTOS+=("${proto}")
+  done <  <(find "${API_ROOT}" -name '*.proto' -print0)
+
+  # Lint and exit if the lint argument is provided.
+  if [[ "${1-}" == "lint"* ]]; then
+    BUF_LINT_CONFIG=$(cat "${REPO_ROOT}/buf.json")
+
+    if [[ ! -f "${CLANG_FORMAT_BIN}" ]]; then
+      echo "error: clang-format not available, please run \`yarn install\`."
+      exit 1
+    fi
+
+    set +e  # Count the failures ourselves instead of early exit so user can see all lint failures.
+    lint_ok=true
+
+    # buf lint. errors are not auto-fixable, so we output the issues with them regardless.
+    for proto in "${PROTOS[@]}"; do
+      if ! output=$(protoc_lint "${proto}"); then
+        echo "--- ${proto}"
+        echo "${output}" | sed 's/--buf-check-lint_out: //' | cut -d":" -f2-
+        lint_ok=false
+      fi
+    done
+
+    # clang-format for whitespace.
+    if [[ "${1-}" == "lint:fix" ]]; then
+      # Fix whitespace in protos.
+      for proto in "${PROTOS[@]}"; do
+        "${CLANG_FORMAT_BIN}" -i "${proto}"
+      done
+    else
+      # Check for diffs in formatted and existing proto.
+      for proto in "${PROTOS[@]}"; do
+        if ! output=$("${CLANG_FORMAT_BIN}" "${proto}" | diff -u "${proto}" -); then
+          echo "${output}"
+          lint_ok=false
+        fi
+      done
+    fi
+
+    ${lint_ok} && exit 0 || exit 1
+  fi
+
+  echo "Compiling protos using $("${PROTOC_BIN}" --version) for protoc..."
+  for proto in "${PROTOS[@]}"; do
+    protoc_command "${proto}"
+  done
+}
+
+# Usage md5check <desired_md5> <filename>.
+md5check() {
+  desired_md5="${1}"
+    filename="${2}"
+  file_md5=$(md5sum "${filename}" | cut -d' ' -f1)
+
+  if [[ "${file_md5}" != "${desired_md5}" ]]; then
+    echo "error: MD5 sum of '${filename}' was ${file_md5} instead of ${desired_md5}."
+    exit 1
+  fi
+}
+
+### Set up dependencies.
+mkdir -p "${GOBIN}"
+
+# Install clang-format.
+CLANG_FORMAT_BIN="${GOBIN}/clang-format"
+if [[ ! -f "${CLANG_FORMAT_BIN}" ]]; then
+  echo "info: Downloading clang-format to build environment"
+
+  case "${OSTYPE}" in
+    "darwin"*) clang_format_os="darwin"; clang_format_md5="c3ebe742599dcc38b9dc6544cacd69bb" ;;
+    "linux"*) clang_format_os="linux"; clang_format_md5="fee8c52e196e28ae5928d6ff8757f58c" ;;
+    *) echo "error: Unsupported OS '${OSTYPE}' for clang-format install, please install manually" && exit 1 ;;
+  esac
+
+  curl -sSL -o "/tmp/clang-format" \
+    "https://github.com/angular/clang-format/raw/v1.4.0/bin/${clang_format_os}_x64/clang-format"
+  md5check "${clang_format_md5}" "/tmp/clang-format"
+  chmod a+x "/tmp/clang-format"
+  mv "/tmp/clang-format" "${CLANG_FORMAT_BIN}"
+
+fi
+
+# Install or update protobuf.
+if [[ -f /etc/alpine-release ]]; then
+  # Alpine must use it's built-in version of protoc since the published version uses glibc and does not support musl.
```

Comment:
do we use alpine docker image to run the script?

Reply:
I removed all the alpine-specific code.